channeldb/codec.go

```diff
@@ -0,0 +1,95 @@
+package channeldb
+
```

Comment:
Ah, I hadn't got to this file when I wrote my comments above about the naming choice. You can safely ignore those comments. 

Excellent work starting this file! In a later PR we can greatly reduce the duplication in the current db encoding/decoding code by expanding the switch statements here with additional cases.